PepperDashEssentials/Room/Types/EssentialsHuddleVtc1Room.cs

```diff
@@ -698,28 +693,35 @@ public static void AllRoomsOff()
 		/// </summary>
 		private void SetCodecExternalSources()
 		{
-			
-			string codecTieLine = "";
-			codecTieLine = ConfigReader.ConfigObject.TieLines.SingleOrDefault(x => x.DestinationKey == VideoCodec.Key).DestinationPort;
-			(VideoCodec as IHasExternalSourceSwitching).ClearExternalSources();
-			(VideoCodec as IHasExternalSourceSwitching).RunRouteAction = RunRouteAction;
-			var srcList =  ConfigReader.ConfigObject.SourceLists.SingleOrDefault(x => x.Key == SourceListKey).Value.OrderBy(kv => kv.Value.Order);;
+			var videoCodecWithExternalSwitching = VideoCodec as IHasExternalSourceSwitching;
+
+			if (videoCodecWithExternalSwitching == null)
+			{
+				return;
+			}
+			else

```

Comment:
For future reference, the else is redundant here. Everything in the `else` block will only happen if `VideoCodecWithExternalSwitching` isn't null, as there is a return statement in the if block.